### ISSUE TITLE
Frida injection via gdbserver

### ISSUE THREAD

**Issue by bannsec** [frida/frida#1217]
Perhaps this is already implemented somewhere.. Basically, it would be very helpful to be able to inject frida via gdbserver. The initial use case I'm thinking is qemu exposing gdb server for other archs, so one could run, for instance, frida on a mips emulated qemu without much change to interaction.

**Comment by oleavr** [comment]
Yes, definitely! We actually already do this on iOS (jailed). Here's the injector:
https://github.com/frida/frida-core/blob/master/src/fruity/injector.vala

And the LLDB client:
https://github.com/frida/frida-core/blob/master/src/fruity/lldb.vala

(Which we could refactor into a generic client that speaks the GDB protocol – e.g. a base class – separating out the LLDB extensions.)

**Comment by blacktop** [comment]
Is there any docs/examples of how to do this injection? Or is it not available yet because it needs "refactor into a generic client that speaks the GDB protocol..." ?

**Comment by oleavr** [comment]
@blacktop It happens transparently e.g. when calling `attach()`, which calls this function:
https://github.com/frida/frida-core/blob/e2a8f91fd35aef6a3567246d2aed0e2f08bda1ca/src/fruity/fruity-host-session.vala#L771-L808

So if you already `spawn()`ed the given PID it will use that LLDB client (A), otherwise it will first try connecting to a device-side frida-server to ask it to `attach()` (B), and if that fails it will start a remote debugserver and do the same as A.
